src/views/Proposals.vue

```diff
@@ -150,10 +159,40 @@ export default {
     }
   },
   methods: {
-    ...mapActions(['getProposals'])
+    ...mapActions(['getProposals']),
+    checkState() {
+      const located = location.hash.replace(/(^#\/yam\/#)/, "");
+      switch (located) {
```

Comment:
This is a lot of code, it could be simplified to one line also it's better to use $router query vars instead of location.hash

Reply:
yes, sent you on chat about this already, we will remove switch to if statement and requires an item list to check from, but also we can just convert it all to the router aspect